### PR TITLE
Avoid using duration::max, cap timeout at a sensible value

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

File: storage/src/vespa/storage/distributor/pendingclusterstate.cpp [hunk 1]
```diff
@@ -22,6 +22,10 @@ using lib::Node;
 using lib::NodeType;
 using lib::NodeState;
 
+namespace {
+    constexpr vespalib::duration MAX_TIMEOUT=3600s;
```
Comment (vekterli) [inline storage/src/vespa/storage/distributor/pendingclusterstate.cpp:26]: Nit: missing spaces

File: storage/src/vespa/storageapi/messageapi/storagecommand.cpp [hunk 2]
```diff
@@ -6,6 +6,10 @@
 
 namespace storage::api {
 
+namespace {
+    constexpr vespalib::duration MAX_TIMEOUT=3600s;
```
Comment (vekterli) [inline storage/src/vespa/storageapi/messageapi/storagecommand.cpp:10]: Nit: missing spaces

**Review by vekterli — APPROVED** [review]
:+1:

File: storage/src/vespa/storageapi/messageapi/storagecommand.cpp [hunk 3]
```diff
@@ -15,10 +19,9 @@ StorageCommand::StorageCommand(const StorageCommand& other)
 
 StorageCommand::StorageCommand(const MessageType& type, Priority p)
     : StorageMessage(type, generateMsgId()),
-        // Default timeout is unlimited. Set from mbus message. Some internal
-        // use want unlimited timeout, (such as readbucketinfo, repair bucket
-        // etc)
-      _timeout(duration::max()),
+        // Default timeout one hour. Set from mbus message. Some internal
+        // use want unlimited timeout, (such as readbucketinfo, repair bucket, etc.)
+      _timeout(MAX_TIMEOUT),
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storageapi/messageapi/storagecommand.cpp:22]: Fixed

File: storage/src/vespa/storage/distributor/pendingclusterstate.cpp (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/distributor/pendingclusterstate.cpp]: Fixed

**Pull request merged**